codalab/apps/web/static/js/worksheet/worksheet_interface.js

```diff
@@ -143,26 +146,54 @@ var Worksheet = React.createClass({
             return false;
         }.bind(this));
     },
-    toggleEditMode: function(editMode) {
+    toggleEditMode: function(editMode, saveChanges) {
         if (editMode === undefined)
           editMode = !this.state.editMode;  // Toggle by default
 
+        if (saveChanges === undefined)
+            saveChanges = true;
+
         if (!editMode) {
           // Going out of raw mode - save the worksheet.
           if (this.canEdit()) {
-            // TODO: This should be done by the editing control in WorksheetItemList 
-            this.state.ws.info.raw = $("#raw-textarea").val().split('\n');
+            // TODO: This should be done by the editing control in WorksheetItemList
+            var editor = ace.edit('worksheet-editor');
```

Comment:
not really. I am afraid that since we are using the CDN, the browser would try to fetch it and error when not find it.